setup.py

```diff
@@ -79,17 +74,83 @@ def install_triton_client():
             f.write(response.content)
         response.close()
         tar_file = tarfile.open(save_name, mode='r')
-        tar_file.extractall()
+        tar_file.extract(f'python/{wheel_file}', path=save_name.parent)
         tar_file.close()
 
-    package_path = save_name.parent / f'python/tensorrtserver-{TRITON_CLIENT_VERSION}-py2.py3-none-linux_x86_64.whl'
+    package_path = save_name.parent / f'python/{wheel_file}'
 
     return package_path
 
 
+# get torch ,torchvision and tensorflow version
+# reference: https://github.com/openvinotoolkit/nncf/blob/develop/setup.py
+# reference: https://www.tensorflow.org/install/source#gpu
+PRECOMPILED_TENSORFLOW_PAIRS = {
+    "cu110": {
+        "tensorflow": "tensorflow-gpu==2.4.0",
+        "tensorflow-serving-api": "tensorflow-serving-api-gpu==2.4.0"
+    },
+    "cu102": {
+        "tensorflow": "tensorflow-gpu==2.3.0",
+        "tensorflow-serving-api": "tensorflow-serving-api-gpu==2.3.0"
+    },
+    "cu101": {
+        "tensorflow": "tensorflow-gpu==2.3.0",
+        "tensorflow-serving-api": "tensorflow-serving-api-gpu==2.3.0"
+    },
+    "cpu": {
+        "tensorflow": "tensorflow==2.3.0",
+        "tensorflow-serving-api": "tensorflow-serving-api==2.3.0"
+    }
+}
+
+CUDA_VERSION = "cpu"
+PYTHON_VER = f'{sys.version_info[0]}{sys.version_info[1]}'
+torch_cuda_version = "cpu"
+tensorflow_cuda_version = "cpu"
+
+
+if "CUDA_HOME" in os.environ:
+    cuda_version_file = os.path.join(os.environ["CUDA_HOME"], "version.txt")
+    if os.path.exists(cuda_version_file):
+        with open(cuda_version_file) as f:
+            CUDA_VERSION = "cu".join(f.readline().strip().split(" ")[-1].split(".")[:2])
+            if CUDA_VERSION not in ["cpu", "cu92", "cu101", "cu102"]:
+                warnings.warn(
+                    f"There is no pre-complied pytorch 1.5.0 with CUDA {CUDA_VERSION}, "
+                    f"and you might need to install pytorch 1.5.0 with CUDA {CUDA_VERSION} from source."
+                )
+            else:
+                torch_cuda_version = CUDA_VERSION
+
+            if CUDA_VERSION not in PRECOMPILED_TENSORFLOW_PAIRS:
+                warnings.warn(
+                    f"There is no pre-complied tensorflow-gpu >=2.1.0 with CUDA {CUDA_VERSION}, "
+                    f"and you might need to install tensorflow-gpu >=2.1.0 with CUDA {CUDA_VERSION} from source."
+                )
+            else:
+                tensorflow_cuda_version = CUDA_VERSION
+
+TORCH_INSTALL_URL = f'https://download.pytorch.org/whl/{torch_cuda_version}/torch-1.5.0' \
```

Comment:
I think Pytorch 1.5.0 do not support Cuda 11.0, you need to check the highest PyTorch support version

Reply:
Yes, cuda11.0 will be replaced with CPU version currently, should we use a higher torch version to support cuda11.0?

Reply:
Better. A100 only supports CUDA >= 11.0